main/motor.h

```diff
@@ -0,0 +1,111 @@
+#ifndef MAIN_MOTOR_H
+#define MAIN_MOTOR_H
+
+#include <misc_utils/physical_size.h>
+
+#include <chrono>
+#include <limits>
+
+#include <bdc_motor.h>
+#include <driver/pulse_cnt.h>
+#include <esp32-hal.h>
+#include <pid_ctrl.h>
+
+enum MotorID : u_int8_t
+{
+    LeftMotor,
+    RightMotor,
+};
+
+class Motor
+{
+public:
+    using Velocity = micromouse::meters_per_second;
+    using Distance = micromouse::meters;
+    using Time = std::chrono::duration<float, std::milli>;
+
+    static constexpr int bdc_mcpwm_timer_resolution_hz = 10'000'000;  // 10MHz, 1 tick = 0.1us
+    static constexpr int bdc_mcpwm_freq_hz = 25'000;                  // 25KHz PWM
+    // maximum value we can set for the duty cycle, in ticks
+    static constexpr int bdc_mcpwm_duty_tick_max = (bdc_mcpwm_timer_resolution_hz / bdc_mcpwm_freq_hz);
+    static constexpr int bdc_encoder_pcnt_high_limit = 1'000;
+    static constexpr int bdc_encoder_pcnt_low_limit = -1'000;
+    // Theoretical max speed = (max_rpm / 60) * (wheel_gear_teeth_N / motor_gear_teeth_N) * wheel_Perimeter
+    static constexpr float max_theoretical_speed = 590.0f / 60.0f * 25.0f / 18.0f * 0.032f * std::numbers::pi_v<float>;
+    static constexpr float max_speed = 1.94386f;                            // [m/s]
+    static constexpr float max_angular_velocity = 2 * max_speed / 9.9e-2f;  // [rad/s]
+    static constexpr float max_acceleration = 9.5f;                         // [m/s^2]
+
+    Motor(
+        gpio_num_t mcpwm_A,
+        gpio_num_t mcpwm_B,
+        gpio_num_t enc_A,
+        gpio_num_t enc_B,
+        MotorID id,
+        bool reversed = false
+    );
+    ~Motor();
+    Motor(const Motor &other) = delete;
+    Motor &operator=(const Motor &other) = delete;
```

Comment:
Rule of ~~3~~5 (delete move too)

```suggestion
    ~Motor();
    Motor(const Motor &) = delete;
    Motor(Motor &&) = delete;
    Motor &operator=(const Motor &) = delete;
    Motor &operator=(Motor &&) = delete;
```

(Also no need to specify parameter names)